### PR TITLE
feat: status update for production plan if material request type is manufacture

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1199693860932069)

### CODE REVIEW

**Comment by nmimsnikhil** [comment]
@MyuddinKhatri please upload 'gif' for better understanding.

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 1]
```diff
@@ -214,6 +214,17 @@ def update_produced_qty(self, produced_qty, production_plan_item):
 		self.set_status()
 		self.db_set('status', self.status)
 
+	def update_produced_qty_in_mr_item(self, produced_qty, material_request_plan_item):
+		for data in self.mr_items:
+			if data.name == material_request_plan_item:
+				data.produced_qty = produced_qty
+				data.db_update()
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py:221]: - Do not update in for loop.
- Update once the for loop is done

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 2]
```diff
@@ -755,7 +767,7 @@ def get_sub_assembly_items(bom_no, bom_data):
 			get_sub_assembly_items(bom_item.get("bom_no"), bom_data)
 
 @frappe.whitelist()
-def update_per_received_and_status_in_production_plan(purchase_receipt):
+def update_per_received_in_production_plan(purchase_receipt):
```
Comment (rvpasari) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py:770]: This is a function called from the frontend. Docstrings should be in format

**Review by rvpasari — APPROVED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.py (hunk 2 above)
Comment (rvpasari) [reply erpnext/manufacturing/doctype/production_plan/production_plan.py:770]: Please correct the details in the docstring - it doesn't reflect what the function is doing

**Review by rvpasari — CHANGES_REQUESTED** [review]

**Review by vishdha — COMMENTED** [review]
@MyuddinKhatri. On cancel purchase receipt it not updating the production plan status it still shows material received. if you need help to connect concern person

**Comment by MyuddinKhatri** [comment]
> @MyuddinKhatri. On cancel purchase receipt it not updating the production plan status it still shows material received. if you need help to connect concern person



screencast-bloomstack_demo_8000-2021 01 08-11_57_34

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 3]
```diff
@@ -786,8 +797,20 @@ def update_per_received_and_status_in_production_plan(purchase_receipt):
...
+
+def update_status_for_production_plan(production_plan):
+	"""
+	Set status for production plan on the basis of material requests.
+
+	Args:
+		production_plan (string): Production Plan name
+	"""	
+	production_plan = frappe.get_doc("Production Plan", production_plan)
+	all_received =[item.per_received for item in production_plan.mr_items if item.material_request_type == "Purchase"]
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]:
```suggestion
	all_received = [item.per_received for item in production_plan.mr_items if item.material_request_type == "Purchase"]
```

File: erpnext/manufacturing/doctype/production_plan/production_plan.py [hunk 4]
```diff
@@ -786,8 +797,20 @@ def update_per_received_and_status_in_production_plan(purchase_receipt):
...
+	produced_qty =[item.produced_qty for item in production_plan.mr_items if item.material_request_type == "Manufacture" and item.requested_qty == item.produced_qty]
```
Comment (hrwX) [inline erpnext/manufacturing/doctype/production_plan/production_plan.py]:
```suggestion
	produced_qty = [item.produced_qty for item in production_plan.mr_items if item.material_request_type == "Manufacture" and item.requested_qty == item.produced_qty]
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/manufacturing/doctype/production_plan/production_plan.py (hunk 1 above)
Comment (MyuddinKhatri) [reply erpnext/manufacturing/doctype/production_plan/production_plan.py:221]: then produced qty is not getting update

Comment (hrwX) [reply erpnext/manufacturing/doctype/production_plan/production_plan.py:221]: Once you update all the rows in child table, then only update the doc.

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**